src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -219,14 +224,14 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                     // Only add a comma after the first param, if any
                     builder.Append(',');
                 }
-                builder.Append(" error=\"invalid_token\", error_description=\"");
-                builder.Append(errorDescription);
+                builder.Append($" error=\"{eventContext.Error ?? "invalid_token"}\", error_description=\"");
+                builder.Append(eventContext.ErrorDescription ?? "An error occurred when validating the access token.");
                 builder.Append('\"');
 
                 Response.Headers.Append(HeaderNames.WWWAuthenticate, builder.ToString());
             }
 
-            return false;
+            return true;
```

Comment:
Good point. Multiple WWW-Authenticate headers is legal, but browsers don't handle it well. We should be appending to a single header value.